app/controllers/passwords_controller.rb

```diff
@@ -1,4 +1,5 @@
 class PasswordsController < ApplicationController
+  before_action :require_password, only: [:edit, :update]
```

Comment:
Please declare require_password method in PasswordsController as a private method. 
No need to declare it in Application controller